qml/InitDSMonitorDialog.qml

```diff
@@ -0,0 +1,425 @@
+                }
+            }
+        }
+    }
+
+    MessageDialog {
+        id: wrongIP
+        title: "Invalid Discovery Server IP"
+        icon: StandardIcon.Warning
+        standardButtons:  isAccepted ? (StandardButton.Retry | StandardButton.Discard) : StandardButton.Retry
+        text: "The inserted Discovery Server IP " + ip + " is not a valid IP."
+        property string ip: ""
+        property bool isAccepted: true
+        onAccepted: {
+            dialogInitMonitor.open()
+        }
+    }
+
+    MessageDialog {
+        id: emptyPort
+        title: "Invalid Discovery Server Port"
+        icon: StandardIcon.Warning
+        standardButtons:  isAccepted ? (StandardButton.Retry | StandardButton.Discard) : StandardButton.Retry
+        text: "No port has been inserted for the Discovery Server IP " + ip + ". " +
+              "Please fill in the ports of all network addresses."
+        property string ip: ""
+        property bool isAccepted: true
+        onAccepted: {
+            dialogInitMonitor.open()
+        }
+    }
+
+    MessageDialog {
+        id: wrongGuid
+        title: "Invalid Discovery Server GUID"
+        icon: StandardIcon.Warning
+        standardButtons:  isAccepted ? (StandardButton.Retry | StandardButton.Discard) : StandardButton.Retry
+        text: "The inserted Discovery Server GUID is not valid."
+        property bool isAccepted: true
+        onAccepted: {
+            dialogInitMonitor.open()
+        }
+    }
+
+    /**
+     * Returns an regular expression for IPv4 string matching
+     */
+    function ipv4Regex() {
+        var ipv4re = /^(((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3})(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/g
+        return ipv4re
+    }
+
+    /**
+     * Returns an regular expression for IPv6 string matching
+     */
+    function ipv6Regex() {
+        var ipv6re = /(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))/gi;
+        return ipv6re
+    }
+
+    /**
+     * Returns an regular expression that matches any string
+     */
+    function matchAllRegex() {
+        var re = /[\s\S]*/
+        return re
+    }
+
+    /**
+     * Function to parse the input parameter of the initialize monitor dialog for Discovery Server
+     */
+    function initDiscoveryServer(isAccepted) {
```

Comment:
Discuss:
Maybe it is better to do this from the C++ side. There is an easy way to show an error message, and I think it would be better to check this on the C++ side or somewhere with access to Fast interfaces (filter transports, set the actual locator, etc.

Reply:
We finally agreed to leave it here as they are user input checks.